### PR TITLE
Bugfix: use lock.extend replace pexpire

### PR DESCRIPTION
**Pull request by laixintao**
monkey patch lua LUA_EXTEND_TO_SCRIPT

close https://github.com/sibson/redbeat/issues/127
close https://github.com/sibson/redbeat/pull/84
close https://github.com/sibson/redbeat/issues/83

### CODE REVIEW

**Comment by laixintao** [comment]
Currently the process will die if it found out that it no longer own the lock:

```
[2019-05-29 23:38:09,365: DEBUG/MainProcess] beat: Extending lock...
[2019-05-29 23:38:09,366: DEBUG/MainProcess] beat: Releasing Lock                                                                [2019-05-29 23:38:09,366: CRITICAL/MainProcess] beat raised exception <class 'redis.exceptions.LockNotOwnedError'>: LockNotOwnedE
rror("Cannot release a lock that's no longer owned")
Traceback (most recent call last):                                                                                                 File "/Users/laixintao/.virtualenvs/redbeat/lib/python3.7/site-packages/celery/beat.py", line 597, in start
    interval = self.scheduler.tick()
  File "/Users/laixintao/Program/redbeat/redbeat/schedulers.py", line 469, in tick                                                   self.lock.extend(int(self.lock_timeout))
  File "/Users/laixintao/.virtualenvs/redbeat/lib/python3.7/site-packages/redis/lock.py", line 246, in extend
    return self.do_extend(additional_time)                                                                                         File "/Users/laixintao/.virtualenvs/redbeat/lib/python3.7/site-packages/redis/lock.py", line 253, in do_extend
    raise LockNotOwnedError("Cannot extend a lock that's"
redis.exceptions.LockNotOwnedError: Cannot extend a lock that's no longer owned
```

I think it would be better to make this process get blocked and trying to reacquiring the lock?

**Comment by laixintao** [comment]
Hi @sibson some tests fails, but it doesn't seem to relevant to this PR. PTAL

**Comment by laixintao** [comment]
I update the lock extend logic and update readme. What do you think?

File: redbeat/schedulers.py [hunk 1]
```diff
@@ -48,6 +48,28 @@
 CELERY_4_OR_GREATER = CELERY_VERSION[0] >= 4
 REDIS_3_OR_GREATER = REDIS_VERSION[0] >= 3
 
+# Copy from:
```
Comment (sibson) [inline redbeat/schedulers.py]: Is this a direct copy or was it modified? if the later let's call out what was changed and why to make future maintenance easier.

**Comment by sibson** [comment]
This generally looks good to me.  I'm taken the approach of exiting on terminal error conditions rather than deal with the complexity of trying to recover.

File: redbeat/schedulers.py (hunk 1 above)
Comment (laixintao) [reply redbeat/schedulers.py]: It was modified and I changed the comment.

**Comment by laixintao** [comment]
Conflict resolved. PTAL @sibson

**Comment by sibson** [comment]
LGTM, thanks.  I'm going to see if we can resolve the failing test on master before merging this.  If that doesn't happen soon, I'll merge this regardless.

**Comment by laixintao** [comment]
ping @sibson

**Pull request merged**